x-pack/plugins/apm/server/lib/search_strategies/failed_transactions_correlations/async_search_service.ts

```diff
@@ -35,10 +36,11 @@ export const asyncErrorCorrelationSearchServiceProvider = (
         includeFrozen,
       };
 
-      const { fieldCandidates } = await fetchTransactionDurationFieldCandidates(
-        esClient,
-        params
-      );
+      const {
+        fieldCandidates: candidates,
+      } = await fetchTransactionDurationFieldCandidates(esClient, params);
+
+      const fieldCandidates = candidates.filter((t) => !(t === EVENT_OUTCOME));
```

Comment:
Can you move this transformation to `fetchTransactionDurationFieldCandidates`?

```suggestion

```

Additionally, since we eventually want this to be an array, can we do away with the `Set` and just use make it an array from the beginning. Currently we use `forEach` and mutate two separate data structures  (`acceptableFields` and `finalFieldCandidates`). I think we can simplify this by using immutable operation (`filter` and `map`) and perhaps only a single data structure.